pilot/pkg/security/authn/policy_applier.go

```diff
@@ -27,7 +27,7 @@ type PolicyApplier interface {
 	// InboundFilterChain returns inbound filter chain(s) for the given endpoint (aka workload) port to
 	// enforce the underlying authentication policy.
 	InboundFilterChain(endpointPort uint32, sdsUdsPath string, node *model.Proxy,
-		listenerProtocol networking.ListenerProtocol) []networking.FilterChain
+		listenerProtocol networking.ListenerProtocol, trustDomainAliases []string) []networking.FilterChain
```

Comment:
I think methods with too many parameters are not a good sign - MeshConfig already has this info ( and the trustDomain), just pass it if needed ( I don't think it is - since it's probably already passed when the filter is created)

Reply:
replacing []string parameter with a new meshconfig parameter does not decrease number of parameters in this case though? PushContext/MeshConfig isn't available where required so this was added